taipy/core/_scheduler/_scheduler.py

```diff
@@ -174,18 +180,21 @@ def __unlock_edit_on_outputs(job):
     @classmethod
     def _on_status_change(cls, job: Job):
         if job.is_finished():
+            if job.is_cancelled():
+                cls.__unblock_jobs(if_canceled=True)
             cls.__unblock_jobs()
             cls.__execute_jobs()
 
     @classmethod
-    def __unblock_jobs(cls):
+    def __unblock_jobs(cls, if_canceled: bool = False):
         for job in cls.blocked_jobs:
             if not cls._is_blocked(job):
                 with cls.lock:
                     try:
-                        job.pending()
+                        if not if_canceled:
```

Comment:
At this point, you don't know if the current job has been submitted with the same submit ID than the one canceled.
What about